internal/backend/sftp/config.go

```diff
@@ -56,10 +58,13 @@ func ParseConfig(s string) (interface{}, error) {
 		host = data[0]
 		dir = data[1]
 		// split user and host at the "@"
-		data = strings.SplitN(host, "@", 2)
+		//
+		// it reverts the host data to support user@domain@hostname notation
+		// that's used by several SFTP servers providers
+		data = strings.SplitN(stringutil.Reverse(host), "@", 2)
 		if len(data) == 2 {
-			user = data[0]
-			host = data[1]
+			user = stringutil.Reverse(data[1])
+			host = stringutil.Reverse(data[0])
 		}
```

Comment:
While I see the nifty trick in reversing the string, it doesn't seem very clean.

Also doesn't feel great importing an *examples* dependency? I haven't seen that before, are they even something one can count in being there and not changing too much? Perhaps it's not too uncommon in the Go world, I have no idea.

What about something like this instead?

~~~
data = strings.SplitN(stringutil.Reverse(host), "@", 3)
if len(data) == 3 {
  user = data[0] + "@" + data[1]
  host = data[2]
} else {
  user = data[0]
  host = data[1]
}
~~~

I'm just thinking it's less magic, easier to read down the road, and one less dependency.

Reply:
Yup, I agree with you.